battle-service/src/main/java/by/ibondarau/tictactoe/battleservice/controller/BattleController.java

```diff
@@ -0,0 +1,113 @@
+@RequestMapping("/battle")
+public class BattleController {
+
+    @GetMapping("/{battleId}")
+    public ResponseEntity<BattleResponseDto> get(@PathVariable Integer battleId) {
+        BattleResponseDto battleResponseDto = new BattleResponseDto();
+        battleResponseDto.setId(battleId);
+        battleResponseDto.setFirstPlayerId(1);
+        battleResponseDto.setSecondPlayerId(2);
+        battleResponseDto.setSize(3);
+        battleResponseDto.setStatus("Finished");
+        battleResponseDto.setMoves(
+                Arrays.asList()
+        );
+        battleResponseDto.setNextMove(null);
+        battleResponseDto.setResult("First wins");
+        battleResponseDto.setCreated(ZonedDateTime.now().toInstant());
+        battleResponseDto.setStarted(ZonedDateTime.now().toInstant());
+        battleResponseDto.setFinished(ZonedDateTime.now().toInstant());
+
+        return ResponseEntity.ok(battleResponseDto);
+    }
+
+    @PostMapping
+    public ResponseEntity<BattleResponseDto> createBattle(@RequestBody CreateBattleDto createBattleDto) {
+        BattleResponseDto battleResponseDto = new BattleResponseDto();
+        battleResponseDto.setId(1);
+        battleResponseDto.setFirstPlayerId(createBattleDto.getUserId());
+        battleResponseDto.setSecondPlayerId(2);
+        battleResponseDto.setSize(3);
+        battleResponseDto.setStatus("Created");
+        battleResponseDto.setMoves(
+                Collections.emptyList()
+        );
+        battleResponseDto.setNextMove(createBattleDto.getMovesFirst());
+        battleResponseDto.setResult(null);
+        battleResponseDto.setCreated(ZonedDateTime.now().toInstant());
+        battleResponseDto.setStarted(null);
+        battleResponseDto.setFinished(null);
+
+        return ResponseEntity.ok(battleResponseDto);
+    }
+
+    @PostMapping("/{battleId}/join")
+    public ResponseEntity<BattleResponseDto> joinBattle(@PathVariable Integer battleId, @RequestBody JoinBattleDto joinBattleDto) {
+        BattleResponseDto battleResponseDto = new BattleResponseDto();
+        battleResponseDto.setId(battleId);
+        battleResponseDto.setFirstPlayerId(1);
+        battleResponseDto.setSecondPlayerId(joinBattleDto.getSecondPlayerId());
+        battleResponseDto.setSize(3);
+        battleResponseDto.setStatus("Started");
+        battleResponseDto.setMoves(
+                Collections.emptyList()
+        );
+        battleResponseDto.setNextMove(1);
+        battleResponseDto.setResult(null);
+        battleResponseDto.setCreated(ZonedDateTime.now().toInstant());
+        battleResponseDto.setStarted(ZonedDateTime.now().toInstant());
+        battleResponseDto.setFinished(null);
+
+        return ResponseEntity.ok(battleResponseDto);
+    }
+
+    @PostMapping("/{battleId}/move")
+    public ResponseEntity<BattleResponseDto> move(@PathVariable Integer battleId, @RequestBody MoveDto nextMove) {
+        BattleResponseDto battleResponseDto = new BattleResponseDto();
```

Comment:
having config which i shared before you will be able to use chains of setters to create instances.
i.e. 
`new BattleResponseDto().setId(battleId).setFirstPlayerId(1)....`